### PR TITLE
[DDW-622] Move currency related code into a dedicated store

### PR DESCRIPTION
**Pull request by daniloprates**
This PR move the currency related code into dedicated store and actions.

## Todos

- [x] Create Currency Store
- [x] Create Currency Actions
- [x] Change the code to point to the new Currency Store and Actions

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Make sure the Currency feature works in the Wallet Summary and Settings > Wallet 

---
## Test scenarios
(Scenarios from the [regression template](https://docs.google.com/spreadsheets/d/1l7khPZ9beM38C0jZTifZO24JK7hIr5E6sfsEwaRfIBs/edit?usp=sharing))

Scenario 1 - Validate "Wallets" tab from "Settings" screen
```gherkin
Given that I am on the "Wallets" tab on the "Settings" screen
When I validate its content
Then I can see a brief description of the feature 
And a switch to enable or disable it
And by default, it's enabled
And "US Dollar" is the default option in the dropdown
```

Scenario 2 - Disable displaying ada on other currencies
```gherkin
Given that a wallet is synced on Daedalus
And I am on the "Wallets" tab on the ""Settings"" screen
When I click on the switch to disable the feature
Then I can see the dropdown and disclaimer are hidden
And on the "Summary" tab for a wallet, no other currency is displayed
```

Scenario 3 - Enable displaying ada on other currencies and check balance of a synced wallet
```gherkin 
Given that a wallet is synced on Daedalus
And the option of displaying ada in other currencies is enabled
When I validate the balance of the wallet in the ""Summary"" tab
Then I can see the balance is displayed in ada
And to the right, the balance is displayed on the selected currency
```

Scenario 4 - Validate selection in "Wallets" tab from "Settings" screen is persisted after restart
```gherkin
Given that I am on the "Wallets" tab on the "Settings" screen
When I enable or disable the feature
And I restart Daedalus
Then I can see that the change was persisted
```
---
### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Review by ManusMcCole — APPROVED** [review]

**Comment by gabriela-ponce** [comment]
@daniloprates I found an issue. It's not very important, but I was able to reproduce it a few times on different platforms:

- After transplanting the chain or the wallets folder on build 17835, for a moment, the currency dropdown shows no results. It fixes itself after a while and no errors on the console were displayed. Check [this screenshot](https://drive.google.com/file/d/1qCSZTaiSJnYhjWymfwmsCIy9L4FSCxy-/view?usp=sharing).

**Comment by daniloprates** [comment]
> @daniloprates I found an issue. It's not very important, but I was able to reproduce it a few times on different platforms:
> 
> * After transplanting the chain or the wallets folder on build 17835, for a moment, the currency dropdown shows no results. It fixes itself after a while and no errors on the console were displayed. Check [this screenshot](https://drive.google.com/file/d/1qCSZTaiSJnYhjWymfwmsCIy9L4FSCxy-/view?usp=sharing).

Thanks @gabriela-ponce. This will be fixed in [this card](https://jira.iohk.io/browse/DDW-629)

File: source/renderer/app/stores/CurrencyStore.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+export default class CurrencyStore extends Store {
+  @observable currencyIsFetchingList: boolean = false;
+  @observable currencyIsFetchingRate: boolean = false;
+  @observable currencyIsAvailable: boolean = false;
+  @observable currencyIsActive: boolean = false;
+  @observable currencyList: Array<Currency> = [];
+  @observable currencySelected: ?Currency = null;
+  @observable currencyRate: ?number = null;
+  @observable currencyLastFetched: ?Date = null;
```
Comment (DominikGuzei) [inline source/renderer/app/stores/CurrencyStore.js:20]: I think we could remove `currency…` at the beginning of these store properties. And reference it directly from the store like this: `currency.rate` in the components

**Review by DominikGuzei — CHANGES_REQUESTED** [review]
Great work, code looks clean 🎉 I just added a small suggestion

File: source/renderer/app/stores/CurrencyStore.js (hunk 1 above)
Comment (daniloprates) [reply source/renderer/app/stores/CurrencyStore.js]: Nice catch! Done ✅

**Review by nikolaglumac — APPROVED** [review]
Nice work @daniloprates 👌

**Pull request merged**